alluxio/alluxio_file_system.py

```diff
@@ -674,3 +674,510 @@ def _validate_path(self, path):
+    ALLUXIO_PAGE_SIZE_DEFAULT_VALUE = "1MB"
+    ALLUXIO_SUCCESS_IDENTIFIER = "success"
+    LIST_URL_FORMAT = "http://{worker_host}:{http_port}/v1/files"
+    FULL_PAGE_URL_FORMAT = (
+        "http://{worker_host}:{http_port}/v1/file/{path_id}/page/{page_index}"
+    )
+    PAGE_URL_FORMAT = "http://{worker_host}:{http_port}/v1/file/{path_id}/page/{page_index}?offset={page_offset}&length={page_length}"
+    WRITE_PAGE_URL_FORMAT = (
+        "http://{worker_host}:{http_port}/v1/file/{path_id}/page/{page_index}"
+    )
+    GET_FILE_STATUS_URL_FORMAT = "http://{worker_host}:{http_port}/v1/info"
+    LOAD_SUBMIT_URL_FORMAT = (
+        "http://{worker_host}:{http_port}/v1/load?path={path}&opType=submit"
+    )
+    LOAD_PROGRESS_URL_FORMAT = (
+        "http://{worker_host}:{http_port}/v1/load?path={path}&opType=progress"
+    )
+    LOAD_STOP_URL_FORMAT = (
+        "http://{worker_host}:{http_port}/v1/load?path={path}&opType=stop"
+    )
+
+    def __init__(
+        self,
+        etcd_hosts=None,
+        worker_hosts=None,
+        options=None,
+        logger=None,
+        http_port="28080",
+        loop=None,
+    ):
+        """
+        Inits Alluxio file system.
+
+        Args:
+            etcd_hosts (str, optional):
+                The hostnames of ETCD to get worker addresses from
+                The hostnames in host1,host2,host3 format. Either etcd_hosts or worker_hosts should be provided, not both.
+            worker_hosts (str, optional):
+                The worker hostnames in host1,host2,host3 format. Either etcd_hosts or worker_hosts should be provided, not both.
+            options (dict, optional):
+                A dictionary of Alluxio property key and values.
+                Note that Alluxio Python API only support a limited set of Alluxio properties.
+            logger (Logger, optional):
+                A logger instance for logging messages.
+            http_port (string, optional):
+                The port of the HTTP server on each Alluxio worker node.
+        """
+        if etcd_hosts is None and worker_hosts is None:
+            raise ValueError(
+                "Must supply either 'etcd_hosts' or 'worker_hosts'"
+            )
+        if etcd_hosts and worker_hosts:
+            raise ValueError(
+                "Supply either 'etcd_hosts' or 'worker_hosts', not both"
+            )
+        self.logger = logger or logging.getLogger("AlluxioFileSystem")
+        self.session = None
+
+        # parse options
+        page_size = self.ALLUXIO_PAGE_SIZE_DEFAULT_VALUE
+        if options:
+            if self.ALLUXIO_PAGE_SIZE_KEY in options:
+                page_size = options[self.ALLUXIO_PAGE_SIZE_KEY]
+                self.logger.debug(f"Page size is set to {page_size}")
+        self.page_size = humanfriendly.parse_size(page_size, binary=True)
+        self.hash_provider = ConsistentHashProvider(
+            etcd_hosts, worker_hosts, self.logger
+        )
```

Comment:
Quick question on asyncio here: Although this class uses asyncio (e.g. all HTTP requests are replaced with async), the etcd side (`worker_ring.py`) still uses blocking HTTP requests. Would these two different communication styles interfere with each other?

Reply:
With my shallow knowledge, the etcd client would block for another thread for a while if we use loop.run_in_executor which do affect other async calls. But that's inevitable if we have blocking calls...unless we use https://github.com/martyanov/aetcd or make our own fork or even use rust binding to rewrite the whole iopath T T

Reply:
Another way to unify the behavior would be using ProcessPoolExecutor
```
 with ProcessPoolExecutor() as executor:
        loop.run_in_executor(executor)
```